x/oracle/keeper/oracle.go

```diff
@@ -0,0 +1,165 @@
+package keeper
+
+import (
+	"fmt"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/medibloc/panacea-core/v2/x/oracle/types"
+	"github.com/tendermint/tendermint/crypto/secp256k1"
+)
+
+// VoteOracleRegistration defines to vote for the new oracle's verification results.
+func (k Keeper) VoteOracleRegistration(ctx sdk.Context, signedVote *types.SignedOracleRegistrationVote) error {
+	if err := signedVote.ValidateBasic(); err != nil {
+		return sdkerrors.Wrap(types.ErrOracleRegistrationVote, err.Error())
+	}
+
+	if k.isMaliciousRequest(ctx, signedVote.OracleRegistrationVote, signedVote.Signature) {
+		// TODO implements request slashing
+		return sdkerrors.Wrap(types.ErrDetectionMaliciousBehavior, "")
+	}
+
+	// Validate the status of panacea to ensure that voting is possible.
+	if err := k.validateOracleRegistrationVote(ctx, signedVote); err != nil {
+		return sdkerrors.Wrap(types.ErrOracleRegistrationVote, err.Error())
+	}
+
+	// When all validations pass, the vote is saved.
+	// If it is an oracle that has already voted, it will be overwritten.
+	if err := k.SetOracleRegistrationVote(ctx, signedVote.OracleRegistrationVote); err != nil {
+		return sdkerrors.Wrap(types.ErrOracleRegistrationVote, err.Error())
+	}
+
+	return nil
+}
+
+// isMaliciousRequest defines to check for malicious requests.
+func (k Keeper) isMaliciousRequest(ctx sdk.Context, vote *types.OracleRegistrationVote, signature []byte) bool {
+	voteBz := k.cdc.MustMarshal(vote)
+	// Verifies that voting requests are signed with oraclePrivKey.
+	ok := secp256k1.PubKey(k.GetParams(ctx).OraclePublicKey).VerifySignature(voteBz, signature)
+
+	return !ok
+}
+
+// validateOracleRegistrationVote defines checking the status of a panacea to ensure that voting is possible.
+func (k Keeper) validateOracleRegistrationVote(ctx sdk.Context, signedVote *types.SignedOracleRegistrationVote) error {
```

Comment:
```suggestion
func (k Keeper) validateOracleRegistrationVote(ctx sdk.Context, signedVote *types.OracleRegistrationVote) error {
```

We can just pass the `types.OracleRegistrationVote` :)

Reply:
Yes. I'll apply this.